hyperdome_server/onion.py

```diff
@@ -176,144 +188,177 @@ def __init__(self, common):
+            )
             torrc_template = torrc_template.replace(
-                '{{control_port}}', str(self.tor_control_port))
+                "{{control_port}}", str(self.tor_control_port)
+            )
             torrc_template = torrc_template.replace(
-                '{{control_socket}}', str(self.tor_control_socket))
+                "{{control_socket}}", str(self.tor_control_socket)
+            )
             torrc_template = torrc_template.replace(
-                '{{cookie_auth_file}}', self.tor_cookie_auth_file)
+                "{{cookie_auth_file}}", self.tor_cookie_auth_file
+            )
             torrc_template = torrc_template.replace(
-                '{{geo_ip_file}}', self.tor_geo_ip_file_path)
+                "{{geo_ip_file}}", self.tor_geo_ip_file_path
+            )
             torrc_template = torrc_template.replace(
-                '{{geo_ipv6_file}}', self.tor_geo_ipv6_file_path)
+                "{{geo_ipv6_file}}", self.tor_geo_ipv6_file_path
+            )
             torrc_template = torrc_template.replace(
-                '{{socks_port}}', str(self.tor_socks_port))
+                "{{socks_port}}", str(self.tor_socks_port)
+            )
 
-            with open(self.tor_torrc, 'w') as f:
+            with open(self.tor_torrc, "w") as f:
                 f.write(torrc_template)
 
                 # Bridge support
-                if self.settings.get('tor_bridges_use_obfs4'):
-                    f.write('ClientTransportPlugin obfs4 exec {}\n'.format(
-                        self.obfs4proxy_file_path))
-                    with open(self.common.get_resource_path(
-                            'torrc_template-obfs4')) as o:
+                if self.settings.get("tor_bridges_use_obfs4"):
+                    f.write(
+                        "ClientTransportPlugin obfs4 exec {}\n".format(
+                            self.obfs4proxy_file_path
+                        )
+                    )
+                    with open(
+                        self.common.get_resource_path("torrc_template-obfs4")
+                    ) as o:
                         for line in o:
                             f.write(line)
-                elif self.settings.get('tor_bridges_use_meek_lite_azure'):
-                    f.write('ClientTransportPlugin meek_lite exec {}\n'.format(
-                        self.obfs4proxy_file_path))
-                    with open(self.common.get_resource_path(
-                            'torrc_template-meek_lite_azure')) as o:
+                elif self.settings.get("tor_bridges_use_meek_lite_azure"):
+                    f.write(
+                        "ClientTransportPlugin meek_lite exec {}\n".format(
+                            self.obfs4proxy_file_path
+                        )
+                    )
+                    with open(
+                        self.common.get_resource_path("torrc_template-meek_lite_azure")
+                    ) as o:
```

Comment:
I love these hanging indents, I didn't know you could close them like this (on the same column as the opening part)

Especially cuz it resolves the problem of being unable to have a hanging indent before an indented block

Reply:
yeah it's pretty clean